tests/test_aggr_regressions.py

```diff
@@ -0,0 +1,130 @@
+"""
+Regression tests for issues 386, 449, 484, and 558 — all related to processing complex aggregations and projections.
+"""
+
+import pytest
+import datajoint as dj
+from . import PREFIX
+import uuid
+from .schema_uuid import Topic, Item, top_level_namespace_id
+from .schema_aggr_regress import R, Q, S, A, B, X, LOCALS_AGGR_REGRESS
+
+
+@pytest.fixture(scope="function")
+def schema_aggr_reg(connection_test):
+    context = LOCALS_AGGR_REGRESS
+    schema = dj.Schema(
+        PREFIX + "_aggr_regress",
+        context=context,
+        connection=connection_test,
+    )
+    schema(R)
+    schema(Q)
+    schema(S)
+    yield schema
+    schema.drop()
+
+
+@pytest.fixture(scope="function")
+def schema_aggr_reg_with_abx(connection_test):
+    context = LOCALS_AGGR_REGRESS
+    schema = dj.Schema(
+        PREFIX + "_aggr_regress_with_abx",
+        context=context,
+        connection=connection_test,
+    )
+    schema(R)
+    schema(Q)
+    schema(S)
+    schema(A)
+    schema(B)
+    schema(X)
+    yield schema
+    schema.drop()
+
+
+def test_issue386(schema_aggr_reg):
+    """
+    --------------- ISSUE 386 -------------------
+    Issue 386 resulted from the loss of aggregated attributes when the aggregation was used as the restrictor
+        Q & (R.aggr(S, n='count(*)') & 'n=2')
+        Error: Unknown column 'n' in HAVING
+    """
+    result = R.aggr(S, n="count(*)") & "n=10"
+    result = Q & result
+    result.fetch()
+
+
+def test_issue449(schema_aggr_reg):
+    """
+    ---------------- ISSUE 449 ------------------
+    Issue 449 arises from incorrect group by attributes after joining with a dj.U()
+    """
+    result = dj.U("n") * R.aggr(S, n="max(s)")
+    result.fetch()
+
+
+def test_issue484(schema_aggr_reg):
+    """
+    ---------------- ISSUE 484 -----------------
+    Issue 484
+    """
+    q = dj.U().aggr(S, n="max(s)")
+    n = q.fetch("n")
+    n = q.fetch1("n")
+    q = dj.U().aggr(S, n="avg(s)")
+    result = dj.U().aggr(q, m="max(n)")
+    result.fetch()
+
+
+def test_union_join(schema_aggr_reg_with_abx):
+    """
+    This test fails if it runs after TestIssue558.
```

Comment:
TODO: Document this as a known GitHub Issue after we merge `dev-tests` to trunk.